src/corehost/corehost.cpp

```diff
@@ -78,24 +101,68 @@ int run(const int argc, const pal::char_t* argv[])
         trace::error(_X("A fatal error occurred, the required library %s could not be found at %s"), LIBFXR_NAME, own_dir.c_str());
         return StatusCode::CoreHostLibMissingFailure;
     }
+
     if (!pal::load_library(fxr_path.c_str(), &fxr))
     {
         trace::error(_X("The library %s was found, but loading it from %s failed"), LIBFXR_NAME, fxr_path.c_str());
-        trace::error(_X(" - Installing .NET Core prerequisites might help resolve this problem."));
+        trace::error(_X("  - Installing .NET Core prerequisites might help resolve this problem."));
         trace::error(_X("     %s"), DOTNET_CORE_URL);
         return StatusCode::CoreHostLibLoadFailure;
     }
 
-    // Obtain entrypoint symbols
+    // Clean up on scope exit.
+    auto unload_proc = [&fxr]() { pal::unload_library(fxr); };
```

Comment:
Why do we need this since the OS would do the needful when cleaning up the process?